### PR TITLE
Added skyloong/Qk21 v1 Number Pad

### PR DESCRIPTION
**Pull request by JZ-Skyloong**
Create a number pad keyboard of skyloong.

## Description

It's a 21 keys Number Pad with RGB matrix effect.  keylayout  is 6x4.  RGB matrix LED is ws2812.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/skyloong/qk21/v1/info.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        "val_steps": 20
+    },
+    "url": "https://github.com/JZ-Skyloong",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6021",
+        "vid": "0x1EA7",
+        "max_power": 250
+    },
+    "default_layouts": ["numpad_6x4"],
```
Comment (dunk2k) [inline keyboards/skyloong/qk21/v1/info.json:101]: "default_layout" isn't present in `data/schemas/keyboard.jsonschema`
```suggestion
    "community_layouts": ["numpad_6x4"],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
I'm note seeing `NUM_LOCK_INDEX` anywhere in source code. Is the intention, of NUM_LOCK_INDEX, to enable and change colour of an LED indicator for NUM_LOCK?

File: keyboards/skyloong/qk21/v1/info.json (hunk 1 above)
Comment (JZ-Skyloong) [reply keyboards/skyloong/qk21/v1/info.json:101]: I have checked the layout level, it seems that there is no 6x4 layout in the community, only 6x4 layout in default. What are your comments and suggestions? @dunk2k

**Comment by JZ-Skyloong** [comment]
> I'm note seeing `NUM_LOCK_INDEX` anywhere in source code. Is the intention, of NUM_LOCK_INDEX, to enable and change colour of an LED indicator for NUM_LOCK?

Yes ,it is.

File: keyboards/skyloong/qk21/v1/info.json (hunk 1 above)
Comment (dunk2k) [reply keyboards/skyloong/qk21/v1/info.json:101]: > I have checked the layout level, it seems that there is no 6x4 layout in the community, only 6x4 layout in default. What are your comments and suggestions?

*Note: the following response is my personal understanding and not official QMK response*

`layout/default` serves as:
1. reference for determining if keyboard's layout matches a community layout
2. Container of keycodes for keymap default_<layoutName> when CI Build is compiling a keyboard that has community layout(s) defined.

`layouts/community` is where community members/contributors can input their keymaps and accompanying code for given community layouts.
For note: Community layouts can be used in conjunction with userspaces.

An analogy:
 Think of `layout/default` as templates and `layout/community` as drawings/forms completed using those templates.

Comment (JZ-Skyloong) [reply keyboards/skyloong/qk21/v1/info.json:101]: @dunk2k Thank you for your insight. I will test the change to community layout to see if there is any problem. If everything is normal, I will adopt your suggestion.

**Review by drashna — APPROVED** [review]

**Comment by JZ-Skyloong** [comment]
What can I do for this PR else?  Would you please merge it now? @dunk2k

File: keyboards/skyloong/qk21/v1/readme.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* Hardware Supported: JK172-RGB(4028)-U PCBs, MEGA32U4-MU 
+* Hardware Availability: http://www.skyloong.com.cn
+
+Make example for this keyboard (after setting up your build environment):
+
+    make qk21/v1:default
+
+Flashing example for this keyboard:
+
+    make qk21/v1:default:flash
```
Comment (fauxpark) [inline keyboards/skyloong/qk21/v1/readme.md:17]:
```suggestion
    make skyloong/qk21/v1:default

Flashing example for this keyboard:

    make skyloong/qk21/v1:default:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/skyloong/qk21/v1/info.json [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    "matrix_pins": {
+        "cols": ["F7", "E6", "C7", "D5"],
+        "rows": ["F0", "F1", "F4", "F5", "F6", "B4"]
+    },
+    "processor": "atmega32u4",
+    "ws2812": {
+        "pin": "B7"
+    },
+    "rgb_matrix":{
+        "driver": "WS2812",
```
Comment (fauxpark) [inline keyboards/skyloong/qk21/v1/info.json:25]:
```suggestion
        "driver": "ws2812",
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/skyloong/qk21/v1/config.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+#ifdef RGB_MATRIX_ENABLE
+#define RGB_MATRIX_LED_COUNT 21
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS  //Enable frame effects
+#define RGB_MATRIX_KEYPRESSES  //Reacts to keypress
+#define RGB_DISABLE_WHEN_USB_SUSPENDED   //turn off effects when suspended
+#endif
+
+
+
```
Comment (fauxpark) [inline keyboards/skyloong/qk21/v1/config.h:17]:
```suggestion
#define RGB_MATRIX_LED_COUNT 21
#define RGB_MATRIX_FRAMEBUFFER_EFFECTS  //Enable frame effects
#define RGB_MATRIX_KEYPRESSES  //Reacts to keypress
#define RGB_DISABLE_WHEN_USB_SUSPENDED   //turn off effects when suspended
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/skyloong/qk21/v1/v1.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+#endif  // NUM_LOCK_INDEX
+
+void suspend_power_down_kb() {
+    rgb_matrix_set_flags(LED_FLAG_NONE);
+    rgb_matrix_set_color_all(0, 0, 0);
+}
+
+void suspend_wakeup_init_kb() {
+    rgb_matrix_set_flags(LED_FLAG_ALL);
+}
```
Comment (drashna) [inline keyboards/skyloong/qk21/v1/v1.c:57]: As per other PR, this isn't needed, as the core code handles this, already.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**